Migrate `acoustic_*.py` scripts by `acoustic.py` subcommands

leplatrem opened the pull request:
This PR comes on top of #569 

* [x] Merge #569 first 
* [ ] Rebase

grahamalama commented on ctms/bin/acoustic.py:
```diff
@@ -0,0 +1,150 @@
+@click.option("--newsletter")
+@click.option("--waitlist")
+@click.pass_context
+def resync(ctx, email_list=None, newsletter=None, waitlist=None):
```
Can we add some typing here and/or rename `email_list` to make it obvious that we're passing a file path and not say `--email-list a@example.com b@example.com ...`. I know you have it in the documentation, and `--help` is helpful too:
```
$ ctms/bin/acoustic.py resync --help
Usage: acoustic.py resync [OPTIONS]

  CTMS command to sync contacts with Acoustic.

Options:
  --email-list FILENAME
  ...
```

but "list" is kind of generic, and got me when I was only looking at the function in the context of the test case you wrote.

Also, knowing that `email_list` is a file object and not a `pathlib.Path` would be helpful.

grahamalama commented on ctms/bin/acoustic.py:
```diff
@@ -0,0 +1,150 @@
+    to_resync = []
+    if email_list:
+        for line in email_list.readlines():
+            to_resync.append(line.decode("utf-8").rstrip().lower())
```
Do we need `line.decode("utf-8")`? I think `readlines` will automatically read the file as a list of strings.

grahamalama commented on tests/unit/test_bin_acoustic_resync.py:
```diff
@@ -0,0 +1,39 @@
+    assert len(dbsession.query(PendingAcousticRecord).all()) > 0
+
+
+def test_main_force_resync_by_email_list(dbsession, sample_contacts):
+    _, some_contact = sample_contacts["maximal"]
+
+    assert len(dbsession.query(PendingAcousticRecord).all()) == 0
+
+    with tempfile.NamedTemporaryFile() as temp:
+        temp.write(some_contact.email.primary_email.encode("utf-8"))
+        temp.flush()
+        temp.seek(0)
```
Pytest's tmpdir could be nice here:

```diff
diff --git a/tests/unit/test_bin_acoustic_resync.py b/tests/unit/test_bin_acoustic_resync.py
index be25191..5ec3aaa 100644
--- a/tests/unit/test_bin_acoustic_resync.py
+++ b/tests/unit/test_bin_acoustic_resync.py
@@ -1,7 +1,4 @@
 """Tests for ctms/bin/acoustic.py resync"""
-
-import tempfile
-
 from ctms.bin.acoustic import do_resync
 from ctms.models import PendingAcousticRecord
 
@@ -24,16 +21,13 @@ def test_main_force_resync_by_waitlist(dbsession, sample_contacts):
     assert len(dbsession.query(PendingAcousticRecord).all()) > 0
 
 
-def test_main_force_resync_by_email_list(dbsession, sample_contacts):
+def test_main_force_resync_by_email_list(dbsession, sample_contacts, tmpdir):
     _, some_contact = sample_contacts["maximal"]
 
     assert len(dbsession.query(PendingAcousticRecord).all()) == 0
 
-    with tempfile.NamedTemporaryFile() as temp:
-        temp.write(some_contact.email.primary_email.encode("utf-8"))
-        temp.flush()
-        temp.seek(0)
-
-        do_resync(dbsession, email_list=temp)
+    f = tmpdir.join("temp.txt")
+    f.write(some_contact.email.primary_email)
+    do_resync(dbsession, email_list=f)
 
     assert len(dbsession.query(PendingAcousticRecord).all()) > 0

```

grahamalama reviewed (commented):
Looking good! A few comments as a first pass.

leplatrem commented:
@grahamalama  I addressed your comments in #569 since this PR is about migrating `acoustic_fields.py` and `acoustic_newsletter_mapping.py`  (ie. the last two commits). I could only do by standing on top of #569 
S

grahamalama reviewed (approved):
Looks like the only big change that's needed is replacing `ctx.obj["dbsession"]` with the `SessionLocal` context manager. 
Good to go after that, I think.

Do these acoustic CRUD commands deserve tests as well?

grahamalama commented:
> We have tests for them already.

Looks like we have tests for some of the crud functions in the CLI, but not all. I don't see tests for:
- `create_acoustic_field`
- `create_acoustic_newsletters_mapping`
- `delete_acoustic_field`
- `delete_acoustic_newsletters_mapping`
- `get_all_acoustic_fields`
- `get_all_acoustic_newsletters_mapping`

Some of those are used in fixtures, but they aren't directly tested
